assets/js/machines/openstack/manage/machine-create.js

```diff
@@ -96,6 +96,18 @@ function check_errors(data, selected_template) {
             count = false;
         }
 
+        // Show Warning if count is > quota
+        if (data['count'] > availablequotavm && groupquotavm !== -1) {
+            creation_error('name-creation-error','You do not have the quota to create ' + data['count'] + ' VMs');
+            count = false;
+        } 
+
+        // Limit numbers of VMs created at once in unlimited quotas
+        if (data['count'] > count_limit && groupquotavm === -1) {
+            creation_error('name-creation-error','You cannot create more than ' + count_limit + ' VMs at one time');
+            count=false;
+        }
+
```

Comment:
It feels like these should be aspects of higher-lever functionality, rather than down here in the OpenStack layer.

Reply:
The ability to create multiple VMs at once is unique to OpenStack.

Reply:
I'm sure it's not... I'm sure we used to let people do it with OpenNebula... :worried:

Reply:
It's not part of the old https://cloud.stfc.ac.uk interface. I could probably add support for it with opennebula, but it's being depreciated so it might not be worth it.

Reply:
We never added it for the OpenNebula version, it was on the list to add at a later date. There was of course the undocumented "feature" that you could click create multiple times before the popup disappeared to create multiple VMs

Reply:
Oh ok 😁